### DISCUSSION TITLE
Need some clarification in using EncodeCipReal

### DISCUSSION THREAD

**Discussion by rajspatel1297** [EIPStackGroup/OpENer#368] [Q&A] [answered]
Hi, I'm using the sample application available under source/src/ports/POSIX/sample_application/sampleapplication.c. 

While using EncodeCipReal function, i'm getting a segmentation fault. I just want to try for a simple structure.

Here is my code for encode. 

```
EipUint8 *buf = gas_ptr(i);

float tmp = csv[bnd[i] + 99];

ENIPMessage enip_gas_message;

memcpy(enip_gas_message.message_buffer, &buf, strlen(buf)+1);
enip_gas_message.used_message_length = 4;
enip_gas_message.current_message_position = &enip_gas_message.message_buffer[0];

EncodeCipReal(&tmp, &enip_gas_message);
```
I want to confirm that is this a proper way to pass enip_gas_message to this function?

**Comment by MartinMelikMerkumians** [comment, marked as answer]
You are providing a pointer to you `EipUint8 `pointer to the `EncodeCipReal` function, so its a pointer-pointer
Taking a look at the functions declaration `void EncodeCipReal(const CipReal *const data, ENIPMessage *const outgoing_message)`, you can easily see that it needs a pointer to an CipReal variable, so a `CipReal*`, while you are providing a `EipUint8**`, which the compiler should warn you if you take a look at the messages
